harness/compareArray.js

```diff
@@ -6,40 +6,34 @@ description: |
 defines: [compareArray]
 ---*/
 
-// @ts-check
-
-function isSameValue(a, b) {
-  if (a === 0 && b === 0) return 1 / a === 1 / b;
-  if (a !== a && b !== b) return true;
-
-  return a === b;
-}
-
-/**
- * @template T
- * @param {T[]} a
- * @param {T[]} b
- */
 function compareArray(a, b) {
   if (b.length !== a.length) {
     return false;
   }
 
   for (var i = 0; i < a.length; i++) {
-    if (!isSameValue(b[i], a[i])) {
+    if (!compareArray.isSameValue(b[i], a[i])) {
       return false;
     }
   }
   return true;
 }
 
-/**
- * @template T
- * @param {T[]} actual
- * @param {T[]} expected
- * @param {string} [message]
- */
+compareArray.isSameValue = function(a, b) {
+  if (a === 0 && b === 0) return 1 / a === 1 / b;
+  if (a !== a && b !== b) return true;
+
+  return a === b;
+};
+
+compareArray.format = function(array) {
+  return `[${array.map(String).join(', ')}]`;
+};
```

Comment:
This PR restores the original and much simpler formatting used for compareArray.